Adding projects to android-l10n (#276)
flodolo/documentation: src/products/android-l10n/adding_projects.md

# Adding a new project to android-l10n

Follow the steps below to add new projects to [android-l10n](https://github.com/mozilla-l10n/android-l10n). You will have to coordinate with android-l10n developers and a technical Localization EPM throughout the process.

## Setting up the project in source and l10n repositories

Before anything happens, check that the new project is properly localizable by following the directions in [this section](https://github.com/mozilla-l10n/documentation/blob/main/src/tools/pontoon/adding_new_project.md#adding-a-new-project-on-pontoon). Once the new project exists in its corresponding source repo and is properly localizable, it can then be added to the `android-l10n` repository.

> An Android developer should create a TOML file in their code repository and under it's corresponding project folder (example with Focus [here](https://github.com/mozilla-mobile/firefox-android/tree/main/focus-android)), as well as as a file containing the source strings (example with Focus [here](https://github.com/mozilla-mobile/firefox-android/blob/c718159480512abf370abcfd5aaf54b1d81074d3/focus-android/app/src/main/res/values/strings.xml)). The `firefox-android` repository contains all [current android projects](https://github.com/mozilla-mobile/firefox-android/tree/main).

Reviewer: ```suggestion
Android developers need to create a TOML file in their code repository and under its corresponding project folder (example for Focus [here](https://github.com/mozilla-mobile/firefox-android/blob/main/focus-android/l10n.toml)), as well as as the file containing the source strings in `res/values/strings.xml` (example with Focus [here](https://github.com/mozilla-mobile/firefox-android/blob/c718159480512abf370abcfd5aaf54b1d81074d3/focus-android/app/src/main/res/values/strings.xml)). The `firefox-android` repository contains all [current android projects](https://github.com/mozilla-mobile/firefox-android/tree/main).
```

Reviewer: What about the list of locales in the config file? Are there limitations?

Author: I don't believe there are any limitations, that I can remember (vs iOS that does for locales we add)
Is that what you were pointing out to call out - or am I missing something? 🤔

Reviewer: Yes, but also that the TOML file needs to have some locales listed to start with.

Author: Ah, right. IIRC we announce a new project and let localizers opt-in beforehand (through our communication channels), and so that we have a list of locales to start with. We also need to check that they are already contained in a-c TOML, and otherwise add them there.
I think that covers it?

Reviewer: (if a-c is required, which might not be a given) But yes, that sounds right.

Author: Thanks for confirming, will make fixes

The `firefox-android` repository currently uses a [single workflow](https://github.com/mozilla-mobile/firefox-android/blob/main/.github/workflows/import-l10n.yml) (GitHub action) to import strings for all projects from `android-l10n`. Sync with mobile developers to make sure that the code to import strings for this project has been added.

> On the `android-l10n` repository side, you will need to set up [import automation](https://github.com/mozilla-l10n/android-l10n/tree/main/.github/workflows) to import strings from the code repository, and the automation will create all needed files in the `mozilla-l10n` repo. You may need to sync up with a technical localization EPM to set this up.

Reviewer: ```suggestion
On the `android-l10n` repository side, you will need to set up [automation](https://github.com/mozilla-l10n/android-l10n/tree/main/.github/workflows) to import strings from the code repository. This automation will be responsible for creating all needed files in the repository. You may need to sync up with a technical localization EPM to set this up.
```

> Once automation is in place, it will extract daily new English strings from the code repository. It's also possible to invoke automation [manually](https://github.com/mozilla-l10n/android-l10n/actions).

Reviewer: ```suggestion
Once automation is in place, it will extract new English strings from the code repository. It should also be possible to invoke automation [manually](https://github.com/mozilla-l10n/android-l10n/actions).
```

"daily" or manual execution depend on how the automation is set up

## Add the project in Pontoon

> Once all of this is done and merged, you can now enable the new project on Pontoon. Instructions for adding new projects can be found [here](https://github.com/mozilla-l10n/documentation/blob/main/src/tools/pontoon/adding_new_project.md#create-the-project).

Reviewer: ```suggestion
Once all of this is done and merged, you can now enable the new project in Pontoon. Instructions for adding new projects can be found [here](https://github.com/mozilla-l10n/documentation/blob/main/src/tools/pontoon/adding_new_project.md#create-the-project).
```

## Setting up the project in source and l10n repositories

Before anything happens, check that the new project is properly localizable by following the directions in [this section](https://github.com/mozilla-l10n/documentation/blob/main/src/tools/pontoon/adding_new_project.md#adding-a-new-project-on-pontoon). Once the new project exists in its corresponding source repo and is properly localizable, it can then be added to the `android-l10n` repository.

Android developers need to create a TOML file in their code repository and under its corresponding project folder (example for Focus [here](https://github.com/mozilla-mobile/firefox-android/blob/main/focus-android/l10n.toml)), as well as as the file containing the source strings in `res/values/strings.xml` (example with Focus [here](https://github.com/mozilla-mobile/firefox-android/blob/c718159480512abf370abcfd5aaf54b1d81074d3/focus-android/app/src/main/res/values/strings.xml)). The `firefox-android` repository contains all [current android projects](https://github.com/mozilla-mobile/firefox-android/tree/main).

> The TOML file should be pre-populated with a list of locales. When a new project is announced, the mobile PM typically reaches out to localizers on existing communication channels, so they can opt-in their locale. If the Android project shares strings with `android-components`, then you need to first make sure the locale code is already listed under its corresponding [TOML file](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/l10n.toml). If not, it will need to be added there as well.

Reviewer: ```suggestion
The TOML file should be pre-populated with a list of locales. When a new project is announced, the mobile PM typically reaches out to localizers on existing communication channels, so they can opt in their locale. If the Android project shares strings with `android-components`, then you need to first make sure the locale code is already listed under its corresponding [TOML file](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/l10n.toml). If not, it will need to be added there as well.
```
